app/src/main/java/fr/gaulupeau/apps/Poche/ui/ReadArticleActivity.java

```diff
@@ -322,25 +337,42 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     @Override
     public boolean dispatchKeyEvent(KeyEvent event) {
-        switch(event.getKeyCode()) {
-            case KeyEvent.KEYCODE_PAGE_UP:
-                scroll(true, screenScrollingPercent, smoothScrolling);
-                return true;
 
-            case KeyEvent.KEYCODE_PAGE_DOWN:
-                scroll(false, screenScrollingPercent, smoothScrolling);
-                return true;
-        }
-
-        if(volumeButtonsScrolling) {
-            switch(event.getKeyCode()) {
-                case KeyEvent.KEYCODE_VOLUME_UP:
+        // only do this once per key press (otherwise the action is done twice)
+        if (event.getAction() == KeyEvent.ACTION_DOWN) {
+            switch (event.getKeyCode()) {
+                case KeyEvent.KEYCODE_PAGE_UP:
                     scroll(true, screenScrollingPercent, smoothScrolling);
                     return true;
 
-                case KeyEvent.KEYCODE_VOLUME_DOWN:
+                case KeyEvent.KEYCODE_PAGE_DOWN:
                     scroll(false, screenScrollingPercent, smoothScrolling);
                     return true;
+
+                case KeyEvent.KEYCODE_CAMERA:
+
+                    disableTouch = !disableTouch;
+
+                    if (disableTouch) {
+                        Toast.makeText(this, "Touch screen disabled", Toast.LENGTH_SHORT).show();
+                    } else {
+                        Toast.makeText(this, "Touch screen enabled", Toast.LENGTH_SHORT).show();
+                    }
+                    Log.d(TAG, "toggling touch screen, now disableTouch is " + disableTouch);
+                    return true;
+            }
+
+
+            if (volumeButtonsScrolling) {
```

Comment:
maybe this if block should be merged into the switch-case block before? 
If not, we should also fix here the bug with double scrolling on one button press (handle KeyEvent.ACTION_DOWN).